sale_product_set_variants/wizard/product_set_add.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <record id="product_set_add_form_view" model="ir.ui.view">
+        <field name="name">product.set.add.form.view</field>
+        <field name="model">product.set.add</field>
+        <field name="inherit_id" ref="sale_product_set.product_set_add_form_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='quantity']" position="after">
+                <field name="set_line_ids" widget="one2many">
+                    <tree editable="bottom">
+                        <field name="product_template_id"/>
+                        <field name="product_variant_ids"/>
```

Comment:
the view gets messy when I try to change the variant:
![image](https://user-images.githubusercontent.com/32064796/46339901-3e435800-c634-11e8-8c3b-d44696655ee8.png)
try using the many2many widget